modules/templates/templates/templates.tsx

```diff
@@ -0,0 +1,250 @@
+        return;
+      }
+      updatedTemplate.createdAt = new Date(
+        updatedTemplate.createdAt,
+      ).toISOString();
+    }
+
+    const savedTemplate = await templateService?.updateTemplate(
+      updatedTemplate.id,
+      updatedTemplate,
+    );
+    if (!savedTemplate) return;
+
+    setTemplates(
+      templates.map((t) => (t.id === savedTemplate.id ? savedTemplate : t)),
+    );
+    setEditingTemplate(null);
+    messageHandler.handleSuccess(t("templates.editSuccess"));
+  };
+
+  const handleInputChange = (field: keyof GenesisTemplate, value: string) => {
+    if (!editingTemplate) return;
+
+    setEditingTemplate({ ...editingTemplate, [field]: value });
+  };
+
+  const columns: ColumnConfig<GenesisTemplate>[] = [
+    {
+      title: t("templates.title"),
+      dataIndex: "name",
+      render: (template: GenesisTemplate) =>
+        editingTemplate && editingTemplate.id === template.id ? (
+          <input
+            type="text"
+            value={editingTemplate.name}
+            onChange={(e) => handleInputChange("name", e.target.value)}
+            className={styles.input}
+          />
+        ) : (
+          <div
+            onClick={() =>
+              router.push(`/app/template-detail?id=${template.id}`)
+            }
+            className="flex items-center"
+            style={{ gap: "1vh", cursor: "pointer" }}
+          >
+            <span>{template.name}</span>
+            <FaRegFolderOpen size={17.5} />
+          </div>
+        ),
+    },
+    {
+      title: t("templates.date"),
+      dataIndex: "createdAt",
+      render: (template: GenesisTemplate) =>
+        editingTemplate && editingTemplate.id === template.id ? (
+          <input
+            type="date"
+            value={
+              new Date(editingTemplate.createdAt).toISOString().split("T")[0]
+            }
+            onChange={(e) => handleInputChange("createdAt", e.target.value)}
+            className={styles.input}
+          />
+        ) : (
+          <span>{new Date(template.createdAt).toLocaleDateString()}</span>
+        ),
+    },
+    {
+      title: t("templates.preview"),
+      dataIndex: "preview",
+      render: (template: GenesisTemplate) =>
+        editingTemplate && editingTemplate.id === template.id ? (
+          <input
+            type="text"
+            value={editingTemplate.preview}
+            onChange={(e) => handleInputChange("preview", e.target.value)}
+            className={styles.input}
+          />
+        ) : (
+          <span>{template.preview}</span>
+        ),
+    },
+    {
+      title: t("templates.action"),
+      dataIndex: "id",
+      render: (template: GenesisTemplate) => (
+        <>
+          {editingTemplate && editingTemplate.id === template.id ? (
+            <button onClick={handleSave} className={styles.actionButton}>
+              <FaSave style={{ color: "#59DBBC" }} />
+            </button>
```

Comment:
Don't we have a resource button?
Do you plan to duplicate `styles.actionButton` at any page uses this type of component?

Reply:
These are just icon containers. They're only used in templates and templates-detail and are slightly different

Reply:
They are only used FOR NOW, tomorrow we may use in different places. 

Container is just the folder on the `resource` 
We actually have already an Icon Button we can replace it (because in deed we used them in more places)

Maybe the designer make it little different, but from user experience all the icon buttons shall the same